nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -138,6 +200,65 @@ def __call__(self, imgs):
         return signals, None
 
 
+def _spheres_inversion(seeds_, radius, mask_img, allow_overlap):
+    '''Utility function to create a maps Niimg, containing a map with a sphere
+    of the given radius for each seed_ location.
+
+    Parameters
+    ----------
+    seeds_: List of triplets of coordinates in native space
+        Seed definitions. List of coordinates of the seeds in the same space
+        as target_affine (either of mask or if mask is None MNI space).
+    radius: float
+        Indicates, in millimeters, the radius for the sphere around the seed.
+    mask_img: Niimg-like object
+        NiftiSpheresMasker mask_img, is necessary as reference to back-project 
+        spheres to brain-space.
+    allow_overlap: boolean
+        If False, an error is raised if the maps overlaps (ie at least two
+        maps have a non-zero value for the same voxel).
+
+    Returns
+    -------
+    spheres_map: Niimg-like object
+        A map of each of the sphere's locations. Scaled at places of overlap.
+    '''
+    # Mask is necessary for affine and shape
+    if mask_img is not None:
+        mask = check_niimg_3d(mask_img)
+    else:
+        raise ValueError('Please provide mask_img at initialization to'
+                            ' provide a reference for the inverse_transform')
+
+    flat_mask, _ = masking._load_mask_img(mask_img)
+    flat_mask = flat_mask.reshape(-1) 
+
+    n_seeds = len(seeds_)
+
+    spheres = np.zeros((np.prod(mask.shape), n_seeds))
+
+    for i, sphere in enumerate(_iter_regions_from_spheres(seeds_, radius, 
```

Comment:
I am wondering if we should store the adjacency matrix in the `transform`, to avoid recomputing it here?

Reply:
In my mind, there are some (maybe very special) use-cases where using `.inverse_transform` might used without having used `.transform`. 
For example for visualization purposes only. 
But more problematic:
In the `.transform` the brain image will probably be masked, so that the adjacency will have dimensions `shape = (n_seeds, np.sum(mask_img.get_data())` or `shape=(n_seeds, np.prod(niimg.shape))`. Whereas in the `.inverse_transform` we need dimensions `shape = (n_seeds, np.prod(mask.shape))`.
All cases that would have to be checked and transformed accordingly - which I think will just add some more layers of complexities, that might not be wanted.

Reply:
No, in the `transform` methods, input images are not masked yet.

Reply:
I meant this as a procedure (as in are going to be masked), which affects in the end the size of the `A` or `adjacency` matrix.